utils/orchestrators.ts

```diff
@@ -0,0 +1,120 @@
+import * as df from "durable-functions";
+import { DurableOrchestrationClient } from "durable-functions/lib/src/durableorchestrationclient";
+import { array } from "fp-ts/lib/Array";
+
+import { toError } from "fp-ts/lib/Either";
+import {
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import {
+  IResponseErrorConflict,
+  IResponseErrorInternal,
+  IResponseSuccessAccepted,
+  ResponseErrorConflict,
+  ResponseErrorInternal,
+  ResponseSuccessAccepted
+} from "italia-ts-commons/lib/responses";
+
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { PromiseType } from "italia-ts-commons/lib/types";
+import { StatusEnum as CgnActivatedStatusEnum } from "../generated/definitions/CgnActivatedStatus";
+import { StatusEnum as CgnCanceledStatusEnum } from "../generated/definitions/CgnCanceledStatus";
+import { StatusEnum as CgnPendingStatusEnum } from "../generated/definitions/CgnPendingStatus";
+import { StatusEnum as CgnRevokedStatusEnum } from "../generated/definitions/CgnRevokedStatus";
+import { CgnStatus } from "../generated/definitions/CgnStatus";
+
+/**
+ * The identifier for StartEligibilityCheckOrchestrator
+ * @param fiscalCode the id of the requesting user
+ */
+export const makeUpdateCgnOrchestratorId = (
+  fiscalCode: FiscalCode,
+  cgnStatus: string
+) => `${fiscalCode}-UPDCGN-${cgnStatus}`;
```

Comment:
By doing this, we are stating that a User can have multiple status processing at the same time. Do we have any scenario that requires such?

Reply:
In theory we can have a scenario like this:  a user's CGN is requested for revokation and at the same time the user reaches 36 years old (that triggers cancel operation). It would be a remote possibility but it still exists. BTW I' ve modified `checkUpdateCgnIsRunning` accordingly to cope with this logic so, if you start a request for any status change, let's say a Revokation request, other status change requests (Cancel request for example) will be declined with a conflict response, otherwise other Revokation request are accepted without starting any orchestrator, returning 202 :)

Reply:
Can't we just say that we can handle only one status update at a time? With this principle we can handle both:
* reject a status update
* enqueue a status update

(in case we submit a request while another is processing).

Reply:
Yep, but in this case, we can obtain 202 responses even if a status update is not the same that is processing. Let's say we have a revokation request that is running for a `fiscalCode`. If a cancel request is triggered at the same time we'll obtain 202, because a status update is already running. Placing at the caller side, in this case we think that our cancel request is already running and it's not true.

Reply:
What do you think about this ? @gunzip @BurnedMarshal

Reply:
@AleDore  Can the triggered cancel operation (that has a higher priority) terminate the running (eventually) orchestrator?

Reply:
> @AleDore Can the triggered cancel operation (that has a highest priority) terminate the running (eventually) orchestrator?

Goog point @BurnedMarshal,  IMHO yes